firmware/boron-ins-fsm/src/imDoorSensor.cpp

```diff
@@ -69,6 +70,13 @@ doorData checkIM() {
         // Checks if the 0th bit (counting from 0) of doorStatus is 1
         doorTamperedFlag = (currentDoorData.doorStatus & 0b0001) != 0;
 
+        // Reset timer on receiving a door close message or transition from open to closed + heartbeat
+        if ((currentDoorData.doorStatus & 0b0010) == 0) {
+            if ((currentDoorData.doorStatus & 0b1000) == 0 || (previousDoorData.doorStatus & 0b0010) != 0) {
```

Comment:
When I try to interpret this code, I get:
```
if (door is currently closed) {
   if (door is sent a heartbeat || door was previously open) {
      reset timer
   }
}
```

In other words:
```
If the door is currently closed and it's a heartbeat
OR
if the door is currently closed but was previously open
THEN
reset timer
```

Which isn't the same thing to me as the comment "Reset timer on receiving a door close message or transition from open to closed + heartbeat". This sounds to me like:
```
If the door is currently closed but was previously open
OR
If the door is currently a closed heartbeat but was previous open
THEN
reset timer
```

Am I misunderstanding something? Which is the version that you really wanted to happen?

Reply:
Oops, sorry, I **was** reading it wrong. But I still don't think it matches up with the comment. Now I see that the code is:
```
If the door is currently closed and it's a not a heartbeat
OR
if the door is currently closed but was previously open
THEN
reset timer
```